cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/system/StreamSystemMetadataWriter.java

```diff
@@ -31,16 +31,32 @@
 public class StreamSystemMetadataWriter extends AbstractSystemMetadataWriter {
 
   private final StreamConfig config;
+  private final long creationTime;
+  private final String description;
 
-  public StreamSystemMetadataWriter(MetadataStore metadataStore, Id.Stream streamId, StreamConfig config) {
+  public StreamSystemMetadataWriter(MetadataStore metadataStore, Id.Stream streamId, StreamConfig config,
+                                    @Nullable String description) {
+    this(metadataStore, streamId, config, -1, description);
+  }
+
+  public StreamSystemMetadataWriter(MetadataStore metadataStore, Id.Stream streamId, StreamConfig config,
```

Comment:
Will this be called if Stream properties are updated? in that case will the creation time be updated as to the stream update time?